phoenix-core/src/main/java/org/apache/phoenix/execute/SortMergeJoinPlan.java

```diff
@@ -572,11 +569,11 @@ public MappedByteBufferTupleQueue(int thresholdBytes) {
         }
 
         @Override
-        protected Comparator<MappedByteBufferSegmentQueue<Tuple>> getSegmentQueueComparator() {
-            return new Comparator<MappedByteBufferSegmentQueue<Tuple>>() {
+        protected Comparator<BufferSegmentQueue<Tuple>> getSegmentQueueComparator() {
+            return new Comparator<BufferSegmentQueue<Tuple>>() {
                 @Override
-                public int compare(MappedByteBufferSegmentQueue<Tuple> q1, 
-                        MappedByteBufferSegmentQueue<Tuple> q2) {
+                public int compare(BufferSegmentQueue<Tuple> q1,
+                                   BufferSegmentQueue<Tuple> q2) {
                     return q1.index() - q2.index();
```

Comment:
Think we should eventually get rid of MappedByteBufferSegmentQueue and used the DeferredXXX version instead. Maybe that'll also simply the level of abstraction here. Go with the other changes first and see what you can do here.